### PR TITLE
Definded handling of tag header in a separate TagHandlerInterface

### PR DESCRIPTION
**Pull request by vidarl**
So, instead of dealing with xkeys in both XLocationIdResponseSubscriber and ConfigurableResponseCacheConfigurator I created a interface for it, TagHandlerInterface. Not 100% happy with then name as it might be confusing as FOS also have a TagHandler.

### CODE REVIEW

**Comment by vidarl** [comment]
Please review :  @andrerom @bdunogier, @alongosz

File: src/DependencyInjection/Compiler/KernelPass.php [hunk 1]
```diff
@@ -45,6 +46,14 @@ public function process(ContainerBuilder $container)
         if ($container->getAlias('ezpublish.http_cache.purge_client') == 'ezpublish.http_cache.purge_client.fos') {
             $container->setAlias('ezplatform.http_cache.purge_client', 'ezplatform.http_cache.purge_client.fos');
         }
+
+        $purgeType = $container->getParameter('purge_type');
+        if ($purgeType === 'http') {
+            // Injecting our own Tag handler
+            $definition = $container->getDefinition('fos_http_cache.handler.tag_handler');
+            $definition->setClass('EzSystems\PlatformHttpCacheBundle\Handler\TagHandler');
```
Comment (alongosz) [inline src/DependencyInjection/Compiler/KernelPass.php]: Nitpick: please use `::class`

File: src/Resources/config/services.yml [hunk 2]
```diff
@@ -24,3 +24,12 @@ services:
     ezplatform.http_cache.tag_aware_store:
         class: EzSystems\PlatformHttpCacheBundle\Proxy\TagAwareStore
         arguments: ["%ezplatform.http_cache.store.root%"]
+
+    ezplatform.http_cache.tag_handler:
+        class: EzSystems\PlatformHttpCacheBundle\Handler\TagHandler
+        # use parent: ???
```
Comment (alongosz) [inline src/Resources/config/services.yml]: Hmm, looks like a TODO :)
Answer: in this case probably not as you've overridden constructor to use our custom arguments / services

File: src/ResponseConfigurator/ConfigurableResponseCacheConfigurator.php [hunk 3]
```diff
@@ -32,11 +33,14 @@ class ConfigurableResponseCacheConfigurator implements ResponseCacheConfigurator
      */
     private $defaultTtl;
 
-    public function __construct($enableViewCache, $enableTtlCache, $defaultTtl)
+    private $tagHandler;
```
Comment (alongosz) [inline src/ResponseConfigurator/ConfigurableResponseCacheConfigurator.php]: Nitpick: please add a type hint in PhpDoc

File: src/DependencyInjection/Compiler/KernelPass.php (hunk 1 above)
Comment (vidarl) [reply src/DependencyInjection/Compiler/KernelPass.php]: Fixed

File: src/Resources/config/services.yml (hunk 2 above)
Comment (vidarl) [reply src/Resources/config/services.yml]: Yeah... That line was never meant to be committed.. Fixed.

File: src/ResponseConfigurator/ConfigurableResponseCacheConfigurator.php (hunk 3 above)
Comment (vidarl) [reply src/ResponseConfigurator/ConfigurableResponseCacheConfigurator.php]: fixed

File: src/Handler/TagHandlerInterface.php [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace EzSystems\PlatformHttpCacheBundle\Handler;
+
+use Symfony\Component\HttpFoundation\Response;
+
+interface TagHandlerInterface
```
Comment (andrerom) [inline src/Handler/TagHandlerInterface.php]: Alternative name, which also overlaps with FOS, but with 2.x, is: `ReponseTaggerInterface`
see: http://foshttpcachebundle.readthedocs.io/en/latest/features/tagging.html

Similar to what they do: https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/blob/master/src/Http/SymfonyResponseTagger.php

Comment (vidarl) [reply src/Handler/TagHandlerInterface.php]: Yes, I am aware of that. If there is consensus that ResponeTaggerInterface is a better name, I can change it

Comment (andrerom) [reply src/Handler/TagHandlerInterface.php]: It's fine keeping it like it is for now.
Can you rebase so we can merge?

Comment (vidarl) [reply src/Handler/TagHandlerInterface.php]: Rebased....

**Pull request merged**